robot/packages/cloud_package_manager.go

```diff
@@ -478,15 +479,31 @@ func (m *cloudManager) downloadFileFromGCSURL(
 		return checksum, contentType, err
 	}
 
-	outHash := base64.StdEncoding.EncodeToString(hash.Sum(nil))
-	if outHash != checksum {
+	checksumBytes, err := base64.StdEncoding.DecodeString(checksum)
+	if err != nil {
+		return "", "", errors.Wrap(err, "failed to decode expected checksum")
+	}
+
+	trimmedChecksumBytes := trimLeadingZeroes(checksumBytes)
+	trimmedOutHashBytes := trimLeadingZeroes(hash.Sum(nil))
+
+	if !bytes.Equal(trimmedOutHashBytes, trimmedChecksumBytes) {
 		utils.UncheckedError(os.Remove(downloadPath))
-		return checksum, contentType, errors.Errorf("download did not match expected hash %s != %s", checksum, outHash)
```

Comment:
no objection to having the trimmed hex values in here, but can you include the original inputs as well? if there's an error, we can reproduce our logic, but we can't always recover inputs